helm/config.go

```diff
@@ -192,23 +273,108 @@ type Mapping struct {
 	nodes []namedNode
 }
 
-// NewMapping creates an empty mapping node and initializes shared fields.
-func NewMapping(modifiers ...NodeModifier) *Mapping {
+// NewEmptyMapping creates an empty mapping node and initializes shared fields.
+func NewEmptyMapping(modifiers ...NodeModifier) *Mapping {
+	mapping := &Mapping{}
+	mapping.Set(modifiers...)
+	return mapping
+}
+
+// NewMapping creates a single node mapping node and initializes shared fields.
+func NewMapping(name string, node Node, modifiers ...NodeModifier) *Mapping {
 	mapping := &Mapping{}
 	mapping.Set(modifiers...)
+	mapping.Add(name, node)
+	return mapping
+}
+
+// NewIntMapping creates a new mapping containing a single named integer scalar.
+// The scalar node will not have a comment or block action.
+func NewIntMapping(name string, value int, modifiers ...NodeModifier) *Mapping {
+	mapping := &Mapping{}
+	mapping.Set(modifiers...)
+	mapping.AddInt(name, value)
+	return mapping
+}
+
+// NewStrMapping creates a new mapping node initialzed with string scalars.
+// The mapping and scalar nodes will not have comments or block actions.
+func NewStrMapping(values ...string) *Mapping {
+	mapping := &Mapping{}
+	for i := 0; i < len(values); i += 2 {
+		value := "~"
+		if i+1 < len(values) {
+			value = values[i+1]
+		}
+		mapping.AddStr(values[i], value)
+	}
 	return mapping
 }
 
 // Add a singled named node at the end of the list.
 func (mapping *Mapping) Add(name string, node Node) {
-	mapping.nodes = append(mapping.nodes, namedNode{name: name, node: node})
+	if node != nil {
+		mapping.nodes = append(mapping.nodes, namedNode{name: name, node: node})
+	}
+}
+
+// AddBool adds a named boolean if the value is true.
+func (mapping *Mapping) AddBool(name string, value bool, modifiers ...NodeModifier) {
+	if value {
+		mapping.Add(name, NewScalar("true", modifiers...))
+	}
+}
+
+// AddInt adds a named integer if the value is not 0.
+func (mapping *Mapping) AddInt(name string, value int, modifiers ...NodeModifier) {
+	if value != 0 {
+		mapping.Add(name, NewScalar(strconv.Itoa(value), modifiers...))
+	}
+}
+
+// AddStr adds a named string if the value is not the empty string.
+func (mapping *Mapping) AddStr(name string, value string, modifiers ...NodeModifier) {
+	if value != "" {
+		mapping.Add(name, NewScalar(value, modifiers...))
+	}
+}
+
+// Get returns the named node, or nil if the name cannot be found.
+func (mapping *Mapping) Get(name string) Node {
+	for _, namedNode := range mapping.nodes {
+		if name == namedNode.name {
+			return namedNode.node
+		}
+	}
+	return nil
+}
+
+// Keys returns the the ordered list of node names in the mapping.
+func (mapping *Mapping) Keys() []string {
+	var keys []string
```

Comment:
Pre-allocations feels like a micro-optimization, but okay.

I've renamed the method from `Keys` to `Names` to bring it in line with the general naming scheme. I've used `name` because `helm.Mapping` doesn't support arbitrary keys, but only string *names* (no funny characters or whitespace).